elyra/pipeline/component_catalog.py

```diff
@@ -89,19 +89,28 @@ def check_outstanding_threads(self) -> bool:
             # Attempt to join thread within the given amount of time
             thread.join(timeout=NONBLOCKING_TIMEOUT)
 
+            cumulative_run_time = int(thread.last_warn_time - thread.task_start_time)
```

Comment:
Sorry, but I think this needs to be: `int(time.time() - thread.task_start_time)` otherwise it could be close to `CATALOG_UPDATE_TIMEOUT` seconds short (worst case).

Reply:
Ugh yes, of course! I'm having a true Monday